test/fail_compilation/extra-files/mixin.sh

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+diff -pu --strip-trailing-cr ${EXTRA_FILES}/mixin.d.out test_results/fail_compilation/mixin.d.out
```

Comment:
DO NOT hard-code the test-results folder here!
That's problematic and thus the CIs rightfully fail.
Reading the README often helps (https://github.com/dlang/dmd/tree/master/test#bash-tests), you probably want `${OUTPUT_BASE}` ;-)